### PR TITLE
EMP-356 bind mount qemu arm

### PR DESCRIPTION
**Pull request by oliv3r**

### CODE REVIEW

File: tests/buildenv.sh [hunk 1]
```diff
@@ -9,25 +9,27 @@ result=0
 
 check_filesystem_support()
 {
-    fs="${1}"
-    grep "${fs}" "/proc/filesystems" || result=1
+    for fs in ${FILESYSTEMS}; do
```
Comment (raysiudak) [inline tests/buildenv.sh:74]: Might as well make the FILESYSTEMS constant part of the function then

File: tests/buildenv.sh [hunk 2]
```diff
@@ -9,25 +9,27 @@ result=0
...
+        fi
+
+    done
 }
 
 check_package_installation()
 {
-    cmd="${1}"
-    command -V "${cmd}" || result=1
+    for pkg in ${PACKAGES}; do
```
Comment (raysiudak) [inline tests/buildenv.sh:87]: Same goes for PACKAGES

File: tests/buildenv.sh [hunk 3]
```diff
@@ -2,12 +2,67 @@
...
+BINFMT_MISC="${BINFMT_MISC:-/proc/sys/fs/binfmt_misc/}"
+PACKAGES="apk mksquashfs xz"
 FILESYSTEMS="overlay squashfs tmpfs"
 
+ARMv7_MAGIC="7f454c4601010100000000000000000002002800"
+
 result=0
 
 
+check_emulation_support()
```
Comment (raysiudak) [inline tests/buildenv.sh]: This function is too long and is difficult to understand, could split it up into several self explaining helper functions to improve readability.

File: tests/buildenv.sh [hunk 4]
```diff
@@ -2,12 +2,67 @@
...
+            continue
+        fi
+
+        if grep -q "${ARMv7_MAGIC}" "${emu}"; then
+            __ARM_EMU_BIN="$(grep "interpreter" "${emu}")"
+            __ARM_EMU_BIN="${__ARM_EMU_BIN#interpreter }"
+            break
+        fi
+    done
+
```
Comment (raysiudak) [inline tests/buildenv.sh:44]: remove one newline

File: tests/rootfs.sh [hunk 5]
```diff
@@ -42,6 +48,9 @@ teardown()
         return
     fi
 
+    umount "${rootfs_dir}/${ARM_EMU_BIN}"
+    unlink "${rootfs_dir}/${ARM_EMU_BIN}"
```
Comment (raysiudak) [inline tests/rootfs.sh:52]: is rm not sufficient here?

File: tests/buildenv.sh (hunk 1 above)
Comment (oliv3r) [reply tests/buildenv.sh:74]: We could yes, but you want to have the configurables/main parameters of the script grouped at the top so they are easy to find.

File: tests/buildenv.sh (hunk 4 above)
Comment (oliv3r) [reply tests/buildenv.sh:44]: :+1:

File: tests/rootfs.sh (hunk 5 above)
Comment (oliv3r) [reply tests/rootfs.sh:52]: rm > unlink :p so technically every-time you do an rm, you should wonder, is unlink not sufficient here :)

also, purely coincidentally it aligns nicely :p

**Review by raysiudak — DISMISSED** [review]

**Review by raysiudak — APPROVED** [review]

**Pull request merged**